src/sys/umd.rs

```diff
@@ -0,0 +1,199 @@
+    ///
+    /// # Parameters
+    ///
+    /// - `unit`: The unit to initialise (probably). Should be set to 1.
+    /// - `drive`: A prefix string for the fs device to mount the UMD on (e.g. "disc0:")
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error
+    pub unsafe fn sce_umd_activate(unit: i32, drive: *const u8) -> i32;
+
+    #[psp(0xE83742BA)]
+    /// Deativates the UMD drive
+    ///
+    /// # Parameters
+    ///
+    /// - `unit`: The unit to initialise (probably). Should be set to 1.
+    /// - `drive`: A prefix string for the fs device to mount the UMD on (e.g. "disc0:")
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error
+    pub unsafe fn sce_umd_deactivate(unit: i32, drive: *const u8) -> i32;
+
+    #[psp(0x8EF08FCE)]
+    /// Wait for the UMD drive to reach a certain state
+    ///
+    /// # Parameters
+    ///
+    /// - `state`: UMD state(s) to wait for
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error
+    pub unsafe fn sce_umd_wait_drive_stat(state: UmdStateFlags) -> i32;
+
+    #[psp(0x56202973)]
+    /// Wait for the UMD drive to reach a certain state
+    ///
+    /// # Parameters
+    ///
+    /// - `state`: UMD state(s) to wait for
+    ///
+    /// # Parameters
+    ///
+    /// - `timeout`: Timeout value in microseconds
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error
+    pub unsafe fn sce_umd_wait_drive_stat_with_timer(
+        state: UmdStateFlags,
+        timeout: u32,
+    ) -> i32;
+
+    #[psp(0x4A9E5E29)]
+    /// Wait for the UMD drive to reach a certain state (plus callback)
+    ///
+    /// # Parameters
+    ///
+    /// - `state`: UMD state(s) to wait for
+    ///
+    /// # Parameters
+    ///
+    /// - `timeout`: Timeout value in microseconds
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error
+    pub unsafe fn sce_umd_wait_drive_stat_cb(
+        state: UmdStateFlags,
+        timeout: u32,
+    ) -> i32;
+
+    #[psp(0x6AF9B50A)]
+    /// Cancel a `sce_umd_wait_*` call
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error
+    pub unsafe fn sce_umd_cancel_wait_drive_stat() -> i32;
+
+    #[psp(0x6B4A146C)]
+    /// Get (poll) the current state of the UMD drive
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error, one or more of `UmdStateFlags` on success
+    pub unsafe fn sce_umd_get_drive_stat() -> i32;
+
+    #[psp(0x20628E6F)]
+    /// Get the error code associated with a failed event
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error, the error code on success
+    pub unsafe fn sce_umd_get_error_stat() -> i32;
+
+    #[psp(0xAEE7404D)]
+    /// Register a callback for the UMD drive
+    ///
+    /// # Parameters
+    ///
+    /// - `cbid`: A callback ID created from `sce_kernel_create_callback`.
+    ///   Callback should be of type `UmdCallback`.
+    ///
+    /// # Return Value
+    ///
+    /// < 0 on error
+    pub unsafe fn sce_umd_register_umd_call_back(cbid: i32) -> i32;
```

Comment:
Should this be call_back or callback?

Reply:
The NID is reversed into ` sceUmdRegisterUMDCallBack`, which is not particularly consistent, so there's no clear answer. I'll remove the underscore as it looks better